diff -ru schroot-1.4.0/bin/schroot/schroot-options-base.cc /tmp/SBo/schroot-1.4.0/bin/schroot/schroot-options-base.cc
--- schroot-1.4.0/bin/schroot/schroot-options-base.cc	2009-10-16 04:11:42.759946158 -0400
+++ /tmp/SBo/schroot-1.4.0/bin/schroot/schroot-options-base.cc	2010-02-20 11:11:15.285353671 -0500
@@ -201,7 +201,8 @@
       this->load_sessions = false;
       if (this->chroots.size() != 1 || all_used())
 	throw opt::validation_error
-	  (_("Exactly one chroot must be specified when beginning a session"));
+	  (opt::validation_error::multiple_values_not_allowed,
+	   _("Exactly one chroot must be specified when beginning a session"));
 
       this->all = this->all_chroots = this->all_sessions = false;
     }
@@ -214,7 +215,8 @@
 
       if (!this->session_name.empty())
         throw opt::validation_error
-	  (_("--session-name is not permitted for the specified action; did you mean to use --chroot?"));
+	  (opt::validation_error::invalid_option,
+	   _("--session-name is not permitted for the specified action; did you mean to use --chroot?"));
     }
   else if (this->action == ACTION_HELP ||
 	   this->action == ACTION_VERSION)
@@ -234,7 +236,7 @@
       if (this->all_sessions)
 	this->load_sessions = true;
       if (!this->chroots.empty())
-	throw opt::validation_error(_("--chroot may not be used with --list"));
+	throw opt::validation_error(opt::validation_error::invalid_option, _("--chroot may not be used with --list"));
     }
   else if (this->action == ACTION_INFO ||
 	   this->action == ACTION_LOCATION ||
@@ -259,6 +261,6 @@
       // Something went wrong
       this->load_chroots = this->load_sessions = false;
       this->all = this->all_chroots = this->all_sessions = false;
-      throw opt::validation_error(_("Unknown action specified"));
+      throw opt::validation_error(opt::validation_error::invalid_option, _("Unknown action specified"));
     }
 }
diff -ru schroot-1.4.0/bin/schroot-base/schroot-base-option-action.cc /tmp/SBo/schroot-1.4.0/bin/schroot-base/schroot-base-option-action.cc
--- schroot-1.4.0/bin/schroot-base/schroot-base-option-action.cc	2008-08-22 19:56:52.000000000 -0400
+++ /tmp/SBo/schroot-1.4.0/bin/schroot-base/schroot-base-option-action.cc	2010-02-20 11:07:14.531417058 -0500
@@ -82,7 +82,7 @@
       if (this->current_action == "")
 	this->current_action = action;
       else
-	throw opt::validation_error(_("Only one action may be specified"));
+	throw opt::validation_error(opt::validation_error::multiple_values_not_allowed);
     }
   else
     throw std::logic_error((format(_("%1%: invalid action")) % action).str());
diff -ru schroot-1.4.0/bin/schroot-base/schroot-base-options.cc /tmp/SBo/schroot-1.4.0/bin/schroot-base/schroot-base-options.cc
--- schroot-1.4.0/bin/schroot-base/schroot-base-options.cc	2009-10-16 04:11:42.759946158 -0400
+++ /tmp/SBo/schroot-1.4.0/bin/schroot-base/schroot-base-options.cc	2010-02-20 11:08:41.360230337 -0500
@@ -155,7 +155,7 @@
       else if (this->debug_level == "critical")
 	sbuild::debug_log_level = sbuild::DEBUG_CRITICAL;
       else
-	throw opt::validation_error(_("Invalid debug level"));
+	throw opt::validation_error(opt::validation_error::invalid_option_value, "Invalid debug level");
     }
   else
     sbuild::debug_log_level = sbuild::DEBUG_NONE;
diff -ru schroot-1.4.0/bin/schroot-listmounts/schroot-listmounts-options.cc /tmp/SBo/schroot-1.4.0/bin/schroot-listmounts/schroot-listmounts-options.cc
--- schroot-1.4.0/bin/schroot-listmounts/schroot-listmounts-options.cc	2008-08-22 19:56:52.000000000 -0400
+++ /tmp/SBo/schroot-1.4.0/bin/schroot-listmounts/schroot-listmounts-options.cc	2010-02-20 11:12:46.418418354 -0500
@@ -86,5 +86,5 @@
 
   if (this->action == ACTION_LISTMOUNTS &&
       this->mountpoint.empty())
-    throw opt::validation_error(_("No mount point specified"));
+    throw opt::validation_error(opt::validation_error::at_least_one_value_required, _("No mount point specified"));
 }
diff -ru schroot-1.4.0/bin/schroot-mount/schroot-mount-options.cc /tmp/SBo/schroot-1.4.0/bin/schroot-mount/schroot-mount-options.cc
--- schroot-1.4.0/bin/schroot-mount/schroot-mount-options.cc	2008-08-22 19:56:52.000000000 -0400
+++ /tmp/SBo/schroot-1.4.0/bin/schroot-mount/schroot-mount-options.cc	2010-02-20 11:14:03.573228653 -0500
@@ -99,5 +99,5 @@
 
   if (this->action == ACTION_MOUNT &&
       this->mountpoint.empty())
-    throw opt::validation_error(_("No mount point specified"));
+    throw opt::validation_error(opt::validation_error::at_least_one_value_required,_("No mount point specified"));
 }


 
